src/components/common/Chart/Chart.tsx

```diff
@@ -0,0 +1,170 @@
+import { AnnotationPluginOptions } from 'chartjs-plugin-annotation/types/options';
+
+ChartJS.register(annotationPlugin);
+
+export interface StyleProps {
+  minHeight?: string;
+  maxWidth?: string;
+}
+
+const useStyles = makeStyles<Theme, StyleProps>(() => ({
+  chart: {
+    minHeight: (props) => props.minHeight ?? '200px',
+    maxWidth: (props) => props.maxWidth ?? undefined,
+  },
+}));
+
+export interface ChartProps {
+  type?: keyof ChartTypeRegistry;
+  chartId: string;
+  chartLabels?: string[];
+  chartValues?: number[];
+  customTooltipTitles?: string[];
+  minHeight?: string;
+  maxWidth?: string;
+  barWidth?: number;
+  barAnnotations?: AnnotationPluginOptions;
+  yLimits?: { min?: number; max?: number };
+  showLegend?: boolean;
+}
+
+export default function Chart(props: ChartProps): JSX.Element | null {
+  const { chartLabels, chartValues } = props;
+  const { activeLocale } = useLocalization();
+  const [locale, setLocale] = useState<string | null>(null);
+
+  useEffect(() => {
+    setLocale(activeLocale);
+  }, [activeLocale]);
+
+  if (!locale || !chartLabels || !chartValues) {
+    return null;
+  }
+
+  return (
+    <ChartContent
+      {...props}
+      locale={locale}
+      chartLabels={chartLabels}
+      chartValues={chartValues}
+      key={`${locale}_${chartLabels.length}_${chartValues.length}`}
+    />
+  );
+}
+
+interface ChartContentProps {
+  type?: keyof ChartTypeRegistry;
+  chartId: string;
+  chartLabels: string[];
+  chartValues: number[];
+  customTooltipTitles?: string[];
+  minHeight?: string;
+  maxWidth?: string;
+  locale: string;
+  barWidth?: number;
+  barAnnotations?: AnnotationPluginOptions;
+  yLimits?: { min?: number; max?: number };
+  showLegend?: boolean;
+}
+
+function ChartContent(props: ChartContentProps): JSX.Element {
+  const {
+    type,
+    chartId,
+    chartLabels,
+    chartValues,
+    customTooltipTitles,
+    minHeight,
+    maxWidth,
+    locale,
+    barWidth,
+    barAnnotations,
+    yLimits,
+    showLegend,
+  } = props;
+  const classes = useStyles({ minHeight, maxWidth });
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const chartRef = useRef<ChartJS | null>(null);
+  const theme = useTheme();
+
+  const barThickness = barWidth === undefined ? 50 : barWidth === 0 ? 'flex' : barWidth;
+
+  useEffect(() => {
+    const createChart = async () => {
+      if (chartRef.current) {
+        chartRef.current.destroy();
+        chartRef.current = null;
+      }
+
+      const ctx = canvasRef?.current?.getContext('2d');
+      if (ctx) {
+        const colors = generateTerrawareRandomColors(theme, chartLabels?.length || 0);
+        chartRef.current = await newChart(locale, ctx, {
+          type: type || 'bar',
+          data: {
+            labels: chartLabels,
+            datasets: [
+              {
+                data: chartValues,
+                barThickness,
+                backgroundColor: colors,
+                minBarLength: 3,
```

Comment:
interesting. didn't think about this.. does the barThickness and bar length affect pie chart or are those options ignored?
Not sure how the datasets should be populated based on chart type.. if pie and bar datasets should be different.. maybe they should come from BarChart and PieChart components.
If it works as-is.. we can maybe revisit later.

Reply:
Datasets are the same, array of values and labels. barThickness and barLength are specific to bar chart, they don't affect pie chart though